### PR TITLE
Fix bugs in TensorRT

### PR DESCRIPTION
**Pull request by stevenlix**
1. Add back -Wno-deprecated-declarations flag in onnxruntime_providers.cmake, which is accidentally deleted in another PR
2. Fix a typo in TensorRT EP which causes vector access error in Windows. Also add error check on cudaMemcpy()
3. Update TensorRT parse to the latest

### CODE REVIEW

File: cmake/onnxruntime_providers.cmake [hunk 1]
```diff
@@ -304,6 +304,9 @@ if (onnxruntime_USE_TENSORRT OR onnxruntime_USE_DNNL)
   if(APPLE)
     set_property(TARGET onnxruntime_providers_shared APPEND_STRING PROPERTY LINK_FLAGS "-Xlinker -exported_symbols_list ${ONNXRUNTIME_ROOT}/core/providers/shared/exported_symbols.lst")
   elseif(UNIX)
+    if (onnxruntime_USE_TENSORRT)
+      set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
```
Comment (snnn) [inline cmake/onnxruntime_providers.cmake:308]: Who needs this? Which module?

Comment (jywu-msft) [inline cmake/onnxruntime_providers.cmake:308]: is it safe to override cxx flags globally here? i think previously the hack was done at the top level cmakelists.txt
is the error happening only when building the onnxruntime_providers_tensorrt target?
if so, can we fix it by setting the compile options for that target only?

Comment (stevenlix) [reply cmake/onnxruntime_providers.cmake:308]: TRT reports error,
TensorRT-7.1.3.4/include/NvInfer.h:5565:81: error: ‘IPluginLayer’ is deprecated [-Werror=deprecated-declarations]
         ITensor* const* inputs, int nbInputs, IPluginExt& plugin) TRTNOEXCEPT = 0;

Comment (stevenlix) [reply cmake/onnxruntime_providers.cmake]: The error happened on Linux only. I move the flag to onnxruntime_providers_tensorrt compile option.

**Pull request merged**